File: 25-RunningTimeAndComplexity/Solution.cpp
#include <cmath>
#include <iostream>

using namespace std;

bool isPrime(int n){
    if (n==2) {
        return true;
    } else if (n==1 || n % 2 == 0) {
        return false;
    }

    for (int i=3; i <=sqrt(n); i++) {
        if (n % i == 0) return false;
    }

    return true;
}

int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */   
    int T, n;
    cin >> T;

    for (int i = 0; i < T ; i++) {
        cin >> n;

        if (isPrime(n)) cout << "Prime" << endl;
        else cout << "Not prime" << endl;
    }


    return 0;
}

File: 25-RunningTimeAndComplexity/Solution.py
import math

T = int(input())

def isPrime(n):
    if n == 2:
        return True
    elif n == 1 or n % 2 == 0:
        return False

    for i in range (2, int(math.sqrt(n)+1)):
        if n % i is 0:
            return False

    return True


for _ in range(T):
    n = int(input())

    if n >= 2 and isPrime(n):
        print("Prime")
    else:
        print("Not prime")